apps/client/src/register/components/registerForm.tsx

```diff
@@ -0,0 +1,143 @@
+import { Button } from '@mui/material';
+import FormInput from '../../components/FormInput';
+import React, { useState } from 'react';
+import axios from 'axios';
+
+const RegisterForm: React.FC = () => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+  const [confirmPasswordError, setConfirmPasswordError] = useState('');
+
+  const validatePassword = (password: string) => {
```

Comment:
I saw that you are using `joi` for schema validation, that library could also help you here, handle the submission of the form and pass the data through the schema validation, that way you can make sure all fields are complete and show errors if necessary.